scripts/snakemake_quant_collect.R

```diff
@@ -0,0 +1,46 @@
+## ---------------------------
+##
+## Script name: snakemake_quant_collect.R
+##
+## Description: Combines all quant.sf files into a csv as part of a snakemake pipeline
+##
+## Snakemake inputs: all quant.sf files for one strain (strain controlled by snakemake)
+##                   annotation map from transcript ID to gene name
+##                   output csv file names for NumReads and TPM
+##
+## Author: Taylor Reiter
+##
+## Date Created: 2022-03-11
+##
+## ---------------------------
+
+library(dplyr)
+library(readr)
+library(purrr)
+library(tidyr)
+
+annotation_map <- read_csv(snakemake@input[["annot_map"]], col_names = c("tx_name", "tx_name2", "Name")) %>%
```

Comment:
Where is this input defined?

Reply:
this is defined by the snakemake file. The rule that uses this script looks at the input file for these rules, and then they're passed as objects (lists) to R. it makes it so that if the input file needs to be updated, it only needs to be changed in the snakefile, and those things will be auto-propagated to the scripts that are run by that rule.